src/components/pages/AccountSettings/EditMember.tsx

```diff
@@ -0,0 +1,202 @@
+
+import { RahaState } from "../../../store";
+import { Member } from "../../../store/reducers/members";
+import { getLoggedInMember } from "../../../store/selectors/authentication";
+import { KeyboardAwareScrollContainer } from "../../shared/elements/KeyboardAwareScrollContainer";
+import { Text, Button, TextInput } from "../../shared/elements";
+import { Loading } from "../../shared/Loading";
+import { styles as commonStyles } from "./styles";
+import { getUsername } from "../../../helpers/username";
+import { fonts } from "../../../helpers/fonts";
+import { colors } from "../../../helpers/colors";
+import { editMember } from "../../../store/actions/me";
+import {
+  ApiCallStatus,
+  ApiCallStatusType
+} from "../../../store/reducers/apiCalls";
+import { getStatusOfApiCall } from "../../../store/selectors/apiCalls";
+
+interface OwnProps {}
+
+interface StateProps {
+  loggedInMember: Member;
+  editMemberApiCallStatus?: ApiCallStatus;
+}
+
+interface DispatchProps {
+  editMember: (
+    loggedInMemberId: MemberId,
+    fullName?: string,
+    username?: string
+  ) => void;
+}
+
+type Props = OwnProps & StateProps & DispatchProps & NavigationInjectedProps;
+
+interface State {
+  username: string;
+  fullName: string;
+}
+
+class EditMemberPageView extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    const { loggedInMember } = this.props;
+    this.state = {
+      username: loggedInMember.get("username"),
+      fullName: loggedInMember.get("fullName")
+    };
+  }
+
+  canUpdate() {
+    const { loggedInMember } = this.props;
+    const currentFullName = loggedInMember.get("fullName");
+    const { fullName } = this.state;
+    if (!fullName || currentFullName === fullName) {
+      return false;
+    }
+    return true;
+  }
+
+  onChangeFullName = (newName: string) => {
+    const { loggedInMember } = this.props;
+    const currentUsername = loggedInMember.get("username");
+    const currentFullName = loggedInMember.get("fullName");
+
+    if (newName === currentFullName) {
+      this.setState({
+        fullName: newName,
+        username: currentUsername
+      });
+    } else {
+      this.setState({
+        fullName: newName,
+        username: getUsername(newName)
+      });
+    }
+  };
+
+  onSubmitUpdate = () => {
+    if (this.canUpdate()) {
+      const { loggedInMember, editMember } = this.props;
+      const { fullName, username } = this.state;
+      editMember(loggedInMember.get("memberId"), fullName, username);
+    }
+  };
+
+  render() {
+    const { loggedInMember, navigation, editMemberApiCallStatus } = this.props;
+    if (!loggedInMember) {
+      return <Loading />;
+    }
+
+    const apiCallStatus =
+      editMemberApiCallStatus && editMemberApiCallStatus.status;
+
+    const disableGoBackButton = apiCallStatus === ApiCallStatusType.STARTED;
+    const goBackButtonTitle =
+      apiCallStatus === ApiCallStatusType.SUCCESS ? "Return" : "Cancel";
```

Comment:
nit: Why "Return" instead of "Back"?

Reply:
Hmm.. I think of "Back" as more exiting out of a process and "Return" as what you do when you're done? Fine either way.